lang/syn/src/parser/context.rs

```diff
@@ -167,31 +187,32 @@ impl ParsedModule {
             None => {
                 // The module is referencing some other file, so we need to load that
                 // to parse the items it has.
-                let parent_dir = parent_file.parent().unwrap();
-                let parent_filename = parent_file.file_stem().unwrap().to_str().unwrap();
-                let parent_mod_dir = parent_dir.join(parent_filename);
-
-                let possible_file_paths = vec![
-                    parent_dir.join(format!("{}.rs", item.ident)),
-                    parent_dir.join(format!("{}/mod.rs", item.ident)),
-                    parent_mod_dir.join(format!("{}.rs", item.ident)),
-                    parent_mod_dir.join(format!("{}/mod.rs", item.ident)),
-                ];
-
-                let mod_file_path = possible_file_paths
-                    .into_iter()
-                    .find(|p| p.exists())
-                    .ok_or_else(|| ParseError::new_spanned(&item, "could not find file"))?;
-                let mod_file_content = std::fs::read_to_string(&mod_file_path)
-                    .map_err(|_| ParseError::new_spanned(&item, "could not read file"))?;
-                let mod_file = syn::parse_file(&mod_file_content)?;
+                panic!("not implemented");
```

Comment:
Unrolling macros works for the whole crate (all modules are getting preprocessed and merged into one output with `mod {` scopes), so pointing to one file is pointless here.

I'm going to change the `--file` option to `--crate`, to make it clear that after this change, `anchor idl` command should point to the crate we're going to generate IDL for, rather than a single file.

Reply:
And I'm going to change this part of code accordingly.

Reply:
Fixed. I realized that in our case, we can hit that part of code (content of mod being `None`) only if someone defines an empty `mod` in code. Unlikely, but I'm gracefully handling it just by defining an empty mod.